contracts/proxy/USDTieredSTOProxy.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.4.24;
+
+import "../modules/STO/USDTieredSTOStorage.sol";
+import "./OwnedProxy.sol";
+import "../Pausable.sol";
+import "openzeppelin-solidity/contracts/ReentrancyGuard.sol";
+import "../modules/STO/ISTOStorage.sol";
+import "../modules/ModuleStorage.sol";
+
+/**
+ * @title USDTiered STO module Proxy
+ */
+contract USDTieredSTOProxy is USDTieredSTOStorage, ISTOStorage, ModuleStorage, Pausable, ReentrancyGuard, OwnedProxy {
+
+    /**
+    * @notice Constructor
+    * @param _securityToken Address of the security token
+    * @param _polyAddress Address of the polytoken
+    * @param _implementation representing the address of the new implementation to be set
+    */
+    constructor (address _securityToken, address _polyAddress, address _implementation)
+    public
+    ModuleStorage(_securityToken, _polyAddress)
```

Comment:
Need to initialize `OwnedProxy` as well using `OwnedProxy()`

Reply:
@adamdossa I just noticed that we haven't initialized it in our other proxy modules either. We need to fix this.
We may also consider refactoring owned proxy to use SecurityToken.owner().

Reply:
We don't need to explicitly initialize the `OwnedProxy`. It is similar to `Ownable` contract where there are no arguments in the constructors so it will get automatically initialized. Right now the proxy ownership is given to factory contract which will never able to call any function of the `OwnedProxy` contract. So we could provide the ownership to an issuer by explicitly initializing the `OwnedProxy(SecurityToken)` and change the `_owner()` function definition  to

        function _owner() internal view returns (address) {
                    return IOwnable(SecurityToken).owner();
         }

Reply:
If I remember correctly, the constructor of the inherited contract is executed automatically only if the inheriting contract does not have a constructor of its own. I'll do a quick test to confirm.

Reply:
Inherited constructors are also executing automatically. My bad.

Reply:
It might make sense to just remove the owned part in module proxies for now as we don't support upgrading them (yet)

Reply:
it only used for getting the implementation contract address, transfer proxy ownership, get the owner contract..... So it doesn't affect anything just the addon. so might make sense to remove the `OwnedProxy` contract and move the `__implementation` variable into the `Proxy` contract.

Reply:
The reason I created OwnedProxy was to avoid a possible namespace clash with the `implementation` function (by using the `ifOwner` modifier).

Reply:
In 2.1.0 I can't really see any reason to change the owner to the security token, but in 3.0.0 we'll be re-jigging this stuff to make certain types of upgrade possible (with control split between the factory and issuer). e.g. something like factory can propose an upgrade, issuer actions it.